docs/tutorial/reading_writing.md

```diff
@@ -0,0 +1,336 @@
+  def run: IO[Unit] = ???
+
+end App
+```
+
+Now we can start using the library! First, create a `Path` containing the path to the file you want to read:
+
+```scala mdoc:compile-only
+import fs2.io.file.Path // TODO: delete this and use the Shellfish's one
+val path = Path("testdata/readme.txt")
+```
+
+And use the `read` function to load the file in memory as a string:
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala mdoc:fail
+import cats.effect.{IO, IOApp}
+
+object App extends IOApp.Simple:
+
+  val path = Path("testdata/readme.txt")
+
+  def run: IO[Unit] = path.read
+
+end App
+```
+
+@:choice(static)
+
+```scala mdoc:fail
+import cats.effect.{IO, IOApp}
+
+object App extends IOApp.Simple:
+
+  val path = Path("testdata/readme.txt")
+
+  def run: IO[Unit] = FilesOs.read(path)
+
+end App
+```
+
+@:@
+
+Oops! We got an error saying that the `run` function accepts an `IO[Unit]`, but the `read` function returns an `IO[String]`. This happens because we are not doing anything with the string loaded and therefore not returning `IO[Unit]`. Fortunately, because `IO` is a monad, we can sequence the value obtained by evaluating the `read` computation with another that prints things to the console (and thus returns `IO[Unit]`!). This is achieved by using the `flatMap` function as follows:
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala
+path.read.flatMap(file => IO(println(file)))
+```
+
+@:choice(static)
+
+```scala
+FilesOs.read(path).flatMap(file => IO(println(file)))
+```
+
+@:@
+
+As you might know, what is happening above is that we are calling the `flatMap` method and passing as parameter a function describing what we want to do with the `file` inside the `IO`, in this case, passing it to the computation `IO(println(file))`.
+
+Now pass the program to the `run` method and everything should go nicely:
+
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.syntax.path.*
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object App extends IOApp.Simple:
+
+  val path = Path("testdata/readme.txt")
+
+  def run: IO[Unit] = path.read.flatMap(file => IO(println(file)))
+
+end App
+```
+
+@:choice(static)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.FilesOs
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object App extends IOApp.Simple:
+
+  val path = Path("testdata/readme.txt")
+
+  def run: IO[Unit] = FilesOs.read(path).flatMap(file => IO(println(file)))
+
+end App
+```
+
+@:@
+
+Congratulations! You have just loaded the contents of a file in pure Scala 🎉.
+
+### Exercise
+
+You may not like to keep using the `flatMap` function over and over again to sequence computations. This is why there is the [macro keyword `for`](https://docs.scala-lang.org/scala3/book/control-structures.html#for-expressions) to automatically let the compiler write the `flatMap`s for you. Why don't you try rewriting the program we just did using for-comprehensions? 
```

Comment:
```suggestion
You may not like to keep using the `flatMap` function over and over again to sequence computations. This is why there is the [`for` construct](https://docs.scala-lang.org/scala3/book/control-structures.html#for-expressions) to automatically let the compiler write the `flatMap`s for you. Why don't you try rewriting the program we just did using for-comprehensions? 
```

Reply:
+ 1000. For is definitely not a macro but a compiler built in feature. It's embedded in the language not an external integration.

Reply:
I thought I had changed it before :P I must have forgotten to commit it. Thanks for bringing it back!